SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -380,3 +384,68 @@ void ULayeredLBStrategy::AddStrategyForLayer(const FName& LayerName, UAbstractLB
 	LayerNameToLBStrategy.Add(LayerName, LBStrategy);
 	LayerNameToLBStrategy[LayerName]->Init();
 }
+
+TUniquePtr<SpatialGDK::FLoadBalancingCalculator> ULayeredLBStrategy::CreateLoadBalancingCalculator(FLegacyLBContext& OutCtx) const
+{
+	if (!IsStrategyWorkerAware())
+	{
+		return nullptr;
+	}
+
+	TArray<FName> LayerNames;
+	TArray<TUniquePtr<SpatialGDK::FLoadBalancingCalculator>> Layers;
+	LayerNames.SetNum(LayerNameToLBStrategy.Num());
+	Layers.SetNum(LayerNameToLBStrategy.Num());
+
+	for (auto Entry : LayerNameToLBStrategy)
+	{
+		FName LayerName = Entry.Key;
+		if (const FLayerData* Data = LayerData.Find(LayerName))
+		{
+			LayerNames[Data->LayerIndex] = LayerName;
+			Layers[Data->LayerIndex] = Entry.Value->CreateLoadBalancingCalculator(OutCtx);
+		}
+	}
+
+	auto Calculator = MakeUnique<SpatialGDK::FLayerLoadBalancingCalculator>(LayerNames, MoveTemp(Layers));
+	OutCtx.Layers = Calculator.Get();
+
+	return Calculator;
+}
+
+bool ULayeredLBStrategy::IsStrategyWorkerAware() const
+{
+	for (auto Entry : LayerNameToLBStrategy)
+	{
+		if (!Entry.Value->IsStrategyWorkerAware())
+		{
+			return false;
+		}
+	}
+	return true;
+}
+
+SpatialGDK::FLoadBalancingDecorator* ULayeredLBStrategy::GetLoadBalancingDecorator() const
```

Comment:
Should the decorator object be more compositional, so the children strategies can add their own decorators? If not we should check that none of them attempt to return a valid decorator as it'll be ignored.

Reply:
"children strategy" nope, after thinking about the problem space, I do not want to suggest that using a compositional model is the way to go. Not just yet anyway, there is a lot to figure out before we can make that call.
So I'm going to get rid of the intermediate classes.